### PR TITLE
feat(DepartureList): format display times on the realtime tracking page

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes

**Asana Ticket:** [Departure List | Format Display Time](https://app.asana.com/0/385363666817452/1204591650723169/f)

I tried first reusing `<DepartureTimes />` then just `infoToDisplayTime`, which was almost okay, but I couldn't really figure out how to best adjust that function to accommodate the many differences in UI display logic needed for the realtime tracking page as opposed to the times displayed in the landing page. So this PR creates a new abstraction that I wonder could later be adapted elsewhere.

I basically have a wrapper component for each "row", `<DisplayTime />`, that establishes a new Context containing information on the departure that's conceptually similar to the output of the existing `infoToDisplayTime` function, with slightly different properties. Those properties are accessed by a variety of child components that are responsible for testing distinct smaller pieces of the UI in each row.

image

---

#### General checks
* [x] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [x] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [x] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues. (have checked but not addressed here - have spun out a few other tickets and PRs)

### CODE REVIEW

File: apps/site/assets/ts/helpers/date.ts
Comment (i0sea) [inline apps/site/assets/ts/helpers/date.ts:95]: For schedules this should be `<1 minute away`

Comment (thecristen) [reply apps/site/assets/ts/helpers/date.ts]: Thanks, fixed.

File: apps/site/assets/ts/helpers/date.ts
Comment (i0sea) [inline apps/site/assets/ts/helpers/date.ts:95]: Adding clarity to previous comment, based on the ticket I think this needs a paramter to distinguish if this is a schedule or a prediction. If it's a schedule the text should be "<1 minute away" and if it's a prediction it should be "approaching"

**Review by i0sea — CHANGES_REQUESTED** [review]

File: apps/site/assets/ts/stop/models/displayTimeConfig.ts
Comment (kotva006) [inline apps/site/assets/ts/stop/models/displayTimeConfig.ts:23]: good catch

File: apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx
Comment (kotva006) [inline apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx:77]: Is there a specific reason that `container` is used here to parse the dom instead of the `screen.get/queryBy()` functions?  I think the test would be better if it queried what the display string should be.

File: apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx
Comment (kotva006) [inline apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx:94]: Same here.  `container` should only be used if all other avenues failed.  Such as checking if a specific color gets applied to a background.

File: apps/site/assets/ts/stop/components/DisplayTime.tsx
Comment (kotva006) [inline apps/site/assets/ts/stop/components/DisplayTime.tsx:63]: What is the purpose of relative?  We don't display timezones anywhere and I wonder if this could lead to confusion.  Also talking to Anna in the past we have favored displaying time in the local timezone of the MBTA.

Comment (thecristen) [reply apps/site/assets/ts/stop/components/DisplayTime.tsx:63]: relative is stuff like `3 min`, `Arriving`, etc... as opposed to `4:23 PM`.

File: apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx
Comment (thecristen) [reply apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx:77]: oh!!! yeah I first tried get/queryByRole and it didn't work; I don't think the library supports the `<time>` role yet. and in some cases, I didn't want to query on specific text displayed, because the text might be displayed differently based on many things, whereas the `datetime` attribute will be consistently formatted... so to me that ruled out the get/queryByText family of functions. should I add a `data-testid` for this case maybe?

File: apps/site/assets/ts/stop/components/DisplayTime.tsx
Comment (kotva006) [reply apps/site/assets/ts/stop/components/DisplayTime.tsx:63]: Okay, I mis-understood what this purpose was.

File: apps/site/assets/ts/helpers/date.ts
Comment (thecristen) [reply apps/site/assets/ts/helpers/date.ts:95]: Thanks for raising -- after talking it over with Anna we've decided to keep both prediction and schedule as "<1 minute away" for now.

File: apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx
Comment (thecristen) [reply apps/site/assets/ts/stop/__tests__/DepartureListTest.tsx]: I just pushed a change taking a slightly different approach with this test to avoid needing to use `container`.

**Review by i0sea — APPROVED** [review]
lgtm, after resolving merge conflict 🚀

**Pull request merged**
